### PR TITLE
Adds PersistentVector.mapi

### PR DESCRIPTION
**Pull request by njlr**
This mirrors `List.mapi` and `Seq.mapi` in the core library.

### CODE REVIEW

File: tests/FSharpx.Collections.Tests/PersistentVectorTest.fs [hunk 1]
```diff
@@ -235,11 +235,21 @@ module PersistentVectorTests =
 
                 let vector2 = PersistentVector.map (fun x -> x * 2) (!vector)
 
-                let a = vector2 |> Seq.toArray 
+                let a = vector2 |> Seq.toArray
                 for i in 1..300 do i * 2 |> Expect.equal "map" a.[i-1] }
 
+            test "vector should allow mapi" {
+                let vector = ref PersistentVector.empty
```
Comment (gdziadkiewicz) [inline tests/FSharpx.Collections.Tests/PersistentVectorTest.fs:242]: Why is ref cell needed? Can't we handle it using `mutable`?

Comment (njlr) [reply tests/FSharpx.Collections.Tests/PersistentVectorTest.fs:242]: Copied from the test above "vector with 300 elements should allow pop"

Comment (njlr) [reply tests/FSharpx.Collections.Tests/PersistentVectorTest.fs:242]: Happy to change it to `mutable` (I prefer that too) but was trying to follow style of existing code.

Comment (gdziadkiewicz) [reply tests/FSharpx.Collections.Tests/PersistentVectorTest.fs:242]: I missed that the rest of the file approaches it like that. I would leave it as it is and potentially tackle all of the ref cells in tests in separate PR. @jackfoxy what's your opinion on that?

**Review by gdziadkiewicz — APPROVED** [review]

File: tests/FSharpx.Collections.Tests/PersistentVectorTest.fs (hunk 1 above)
Comment (njlr) [reply tests/FSharpx.Collections.Tests/PersistentVectorTest.fs:242]: OK to merge?

**Comment by njlr** [comment]
Sorry to bump @gdziadkiewicz

**Comment by gdziadkiewicz** [comment]
Feel free to bump me as much as necessary. The problem is that I don't have merge rights for this repo. @forki @mausch @jackfoxy @fsprojectsgit are you able to help with reviewing and merging this PR?

**Comment by njlr** [comment]
Sorry to bump!

**Comment by jackfoxy** [comment]
@dsyme  please remove me from this and any other FsProjects repos I may be as a maintainer.
There are at least a couple of PRs in this repo awaiting review. I don't know if there are any active maintainers on this project now.

**Comment by gdziadkiewicz** [comment]
I'm willing to contribute as an maintainer if an maintainer is needed.

**Comment by dsyme** [comment]
@gdziadkiewicz Lovely, I'll add you as maintainer in this one

@jackfoxy I'll remove you - many thanks for everything you've contributed to this and the other projects over time!!

**Review by gdziadkiewicz — APPROVED** [review]

**Comment by gdziadkiewicz** [comment]
Released in 3.1.0 https://www.nuget.org/packages/FSharpx.Collections/3.1.0

**Pull request merged**